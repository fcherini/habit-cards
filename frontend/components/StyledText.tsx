import { Text, type TextProps } from "react-native";
import { twMerge } from "tailwind-merge";

type ThemedTextTypes =
  | "default"
  | "title"
  | "subtitle"
  | "link"
  | "small"
  | "detail";

export type ThemedTextProps = TextProps & {
  type?: ThemedTextTypes;
};

export function Txt({ type = "default", className, ...rest }: ThemedTextProps) {
  return (
    <Text
      className={twMerge("text-inherit", textTypes[type], className)}
      {...rest}
    />
  );
}

const textTypes: Record<ThemedTextTypes, string> = {
  default: "text-16 leading-none text-inherit",
  small: "text-12 leading-none",
  title: "font-bold text-20 leading-none",
  detail: "text-14 font-semibold",
  subtitle: "font-bold text-16 leading-none",
  link: "text-secondary underline leading-none",
};
