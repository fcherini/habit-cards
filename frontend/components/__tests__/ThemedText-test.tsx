import * as React from "react";
import renderer from "react-test-renderer";

import { Txt } from "../StyledText";

it(`renders correctly`, () => {
  const tree = renderer.create(<Txt>Snapshot test!</Txt>).toJSON();

  expect(tree).toMatchSnapshot();
});
