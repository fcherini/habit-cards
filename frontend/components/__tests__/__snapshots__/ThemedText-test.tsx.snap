// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<Text
  className="leading-none text-base-content text-base"
>
  Snapshot test!
</Text>
`;
